test/test_corpus.cc

```diff
@@ -768,7 +768,28 @@ void testQuickFixCodeActions(LSPWrapper &lspWrapper, Expectations &test, Unorder
 
             UnorderedMap<string, unique_ptr<CodeAction>> receivedCodeActionsByTitle;
             auto &receivedCodeActions = get<vector<unique_ptr<CodeAction>>>(receivedCodeActionResponse);
+            unique_ptr<CodeAction> sourceLevelCodeAction;
+            // One code action should be a 'source' level code action. Remove it.
+            if (!receivedCodeActions.empty()) {
+                for (auto it = receivedCodeActions.begin(); it != receivedCodeActions.end();) {
+                    if ((*it)->kind == CodeActionKind::Source) {
+                        EXPECT_EQ(sourceLevelCodeAction, nullptr) << "Received multiple source-level code actions";
+                        sourceLevelCodeAction = move(*it);
+                        // Remove from vector and continue
+                        it = receivedCodeActions.erase(it);
+                    } else {
+                        it++;
+                    }
+                }
+                EXPECT_NE(sourceLevelCodeAction, nullptr)
+                    << "Expected one source-level code action for code action request";
+            }
+
             for (auto &codeAction : receivedCodeActions) {
+                if (sourceLevelCodeAction != nullptr && codeAction->title == sourceLevelCodeAction->title) {
```

Comment:
Why would this case ever fire if we removed it at L778?

Reply:
EXPECT_EQ doesn't fail the test I don't think. It appends the error and keeps going.

Reply:
Good question. We send two identical versions of the code action with different kinds:

- A `CodeActionKind::Source`, which is what VS Code wants us to use for whole-file fixes.
- A `CodeActionKind::Quickfix`, which is what appears in the quick fix dropdown.

I'm stripping out the `quickfix` variant here.

Reply:
I'll add a comment

Reply:
> EXPECT_EQ doesn't fail the test I don't think. It appends the error and keeps going.

Oooh, right. This is what I forgot. It's been a while since I had to remember the difference between these.